quickwit-storage/src/bundle_storage.rs

```diff
@@ -134,6 +134,14 @@ impl BundleStorageFileOffsets {
 
 #[async_trait]
 impl Storage for BundleStorage {
+    async fn check(&self) -> anyhow::Result<()> {
+        if self.bundle_filepath.exists() {
```

Comment:
The check is incorrect I believe. 
We want the path to exist... in the `Storage`.

Which makes me realize our usage of the Path struct is probably an abuse...

Reply:
What would the right check be there? Perhaps `self.metadata.files.iter().all(|(file, _offset)| file.exists())` to make sure all the referenced files are alive?

Reply:
The bundle storage is  a bit complicated. It is a directory that is encoded in a single file. 
The file itself is in a storage. In other words, bundle_filepath is not something that makes sense on the local filepath.
It is a Path meant to be used with self.storage.

A good low cost fix would be to call
`self.storage.exists(&self.bundle_filepath).await`

It is not an exhaustive check but that will be sufficient. That function will most likely never be called actually.

Reply:
Ok, done.